Gems/Archive/Code/Include/Archive/Clients/ArchiveReaderAPI.h

```diff
@@ -0,0 +1,283 @@
+        ArchiveHeader::TocOffsetU64 m_offset{};
+        //! Span which view of the extracted file data
+        //! If the ArchiveReaderFileSettings specifies decompression should occur,
+        //! Then the extracted file content will be the raw content
+        AZStd::span<AZStd::byte> m_fileSpan;
+        //! Stores any error messages related to extraction of the file from the archive
+        ResultOutcome m_resultOutcome;
+    };
+
+    //! Returns a result structure that indicates if removal of a content file from the
+    //! archive was successful
+    //! Metadata about the file is returned, such as its file path, compressed algorithm ID
+    //! offset from the beginning of the raw file data blocks, uncompressed size and compressed size
+    struct ArchiveListFileResult
+    {
+        //! returns if adding a stream of data to a file within the archive has succeeded
+        //! it does by checking that the ArchiveFileToken != InvalidArchiveFileToken
+        explicit operator bool() const;
+
+        //! The file path of the file being queried
+        AZ::IO::Path m_relativeFilePath;
+        //! Identifier token that allows for quicker lookup of the file in the archive TOC
+        ArchiveFileToken m_filePathToken{ InvalidArchiveFileToken };
+        //! The compression algorithm ID representing the compression algorithm used to store the file
+        Compression::CompressionAlgorithmId m_compressionAlgorithm{ Compression::Uncompressed };
+        //! The uncompressed size of the removed file
+        AZ::u64 m_uncompressedSize{};
+        //! the compressed size of the removed file
+        //! INFO: This value will be a multiple of 512
+        AZ::u64 m_compressedSize{};
+        //! The raw offset of the file in the archive
+        //! As the ArchiveHeader is 512-byte aligned to the beginning of the file
+        //! this value is at least 512,
+        //! NOTE: The TocOffsetU64 structure is used to enforce that the value is >= 512
+        ArchiveHeader::TocOffsetU64 m_offset{};
+        //! Stores error and information messages with listing the contents of the file
+        ResultOutcome m_resultOutcome;
+    };
+
+    //! Encapsulates the result of enumerating files with the archive
+    //! If an error orruces the ResultOutcome unexpected value is set
+    struct EnumerateArchiveResult
+    {
+        //! returns true if the ResultOutcome has a non-error value
+        explicit constexpr operator bool() const;
+        //! Stores error info about enumerating all files within the archive
+        ResultOutcome m_resultOutcome;
+    };
+
+
+    //! Interface for the ArchiveReader of O3DE Archive format
+    //! An ArchiveReaderSettings object can be used to customize the how
+    //! an archive is read.
+    //! Such as the ability to specify the number of read and decompression task that
+    //! can run in parallel
+
+    //! The user can supply there own stream of archive data via the AZ::IO::GenericStream interface
```

Comment:
```suggestion
    //! The user can supply their own stream of archive data via the AZ::IO::GenericStream interface
```

Reply:
Or you are sneakily putting them in to see if anyone is reading them! 😆 But seriously, I really appreciate these comments, very useful!